### PR TITLE
add support for sha384 and sha512 to hashedrekord type

### PR DESCRIPTION
**Pull request by bobcallaway**

### CODE REVIEW

File: pkg/types/hashedrekord/v0.0.1/entry.go [hunk 1]
```diff
@@ -178,11 +179,21 @@ func (v *V001Entry) validate() (pki.Signature, pki.PublicKey, error) {
...
+	default:
+		alg = crypto.SHA256
+	}
+
 	decoded, err := hex.DecodeString(*hash.Value)
 	if err != nil {
 		return nil, nil, err
 	}
-	if err := sigObj.Verify(nil, keyObj, options.WithDigest(decoded)); err != nil {
+	if err := sigObj.Verify(nil, keyObj, options.WithDigest(decoded), options.WithCryptoSignerOpts(alg)); err != nil {
```
Comment (Hayden-IO) [inline pkg/types/hashedrekord/v0.0.1/entry.go:196]: nit for another file: Should we add a comment that the digest in https://github.com/sigstore/rekor/blob/fc28ac1bc2ecae172371df6b63efe48daee863ff/pkg/pki/x509/x509.go#L87 is a default? I was looking through the code and was initially uncertain if that needed to be modified too.

**Review by Hayden-IO — APPROVED** [review]

File: pkg/types/hashedrekord/v0.0.1/entry.go (hunk 1 above)
Comment (bobcallaway) [reply pkg/types/hashedrekord/v0.0.1/entry.go:196]: it is a default, so we should probably add that comment. the `WithCryptoSignerOpts()` already provides a way to override the algorithm.

Comment (ret2libc) [reply pkg/types/hashedrekord/v0.0.1/entry.go:196]: I think I was missing this in my PR #1945 .

**Comment by bobcallaway** [comment]
Closed in deference to #1959

**Pull request closed without merging**